quic/s2n-quic-core/src/endpoint/limits.rs

```diff
@@ -54,8 +57,10 @@ impl Outcome {
     }
 
     /// Cleanly close the connection
-    pub fn close() -> Self {
-        Self::Close
+    pub fn close(reason: &'static [u8]) -> Self {
```

Comment:
I think this is changing public, stable API -- this is re-exported from s2n-quic. You'll want to add a new `pub fn close_with_reason(...)` or similar.